sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioUtils.scala

```diff
@@ -16,92 +16,267 @@
 
 package com.nvidia.spark.rapids
 
+import java.io.FileNotFoundException
+
+import scala.io.Source
+import scala.sys.process.{Process, ProcessLogger}
+
 import org.apache.hadoop.fs.Path
 
+import org.apache.spark.internal.Logging
 import org.apache.spark.sql.catalyst.expressions.{Expression, PlanExpression}
 import org.apache.spark.sql.execution.datasources.{FileIndex, HadoopFsRelation, InMemoryFileIndex}
 import org.apache.spark.sql.execution.datasources.rapids.GpuPartitioningUtils
 
-object AlluxioUtils {
+object AlluxioUtils extends Logging {
+  val mountedBuckets: scala.collection.mutable.Map[String, String] = scala.collection.mutable.Map()
+  var alluxioCmd: Seq[String] = null
+  var alluxioMasterHost: Option[String] = None
+  var alluxio_home: String = "/opt/alluxio-2.8.0"
+  var isInit: Boolean = false
+
+  // Read out alluxio.master.hostname, alluxio.master.rpc.port
+  // from Alluxio's conf alluxio-site.properties
+  // We require an environment variable "ALLUXIO_HOME"
+  // This function will only read once from ALLUXIO/conf.
+  private def initAlluxioInfo(conf: RapidsConf): Unit = {
+    this.synchronized {
+      alluxio_home = scala.util.Properties.envOrElse("ALLUXIO_HOME", "/opt/alluxio-2.8.0")
+      alluxioCmd = conf.getAlluxioCmd.getOrElse(
+        Seq("su", "ubuntu", "-c", s"$alluxio_home/bin/alluxio"))
+
+      if (!isInit) {
+        // Default to read from /opt/alluxio-2.8.0 if not setting ALLUXIO_HOME
+        var alluxio_port: String = "19998"
+        var alluxio_master: String = null
+        var buffered_source: Source = null
+        try {
+          buffered_source = Source.fromFile(alluxio_home + "/conf/alluxio-site.properties")
+          for (line <- buffered_source.getLines) {
+            if (line.startsWith("alluxio.master.hostname")) {
+              alluxio_master = line.split('=')(1).trim
+            } else if (line.startsWith("alluxio.master.rpc.port")) {
+              alluxio_port = line.split('=')(1).trim
+            }
+          }
+        } catch {
+          case e: FileNotFoundException =>
+            throw new RuntimeException(s"Not found Alluxio config in " +
+              s"$alluxio_home/conf/alluxio-site.properties, " +
+              "please check if ALLUXIO_HOME is set correctly")
+        } finally {
+          if (buffered_source != null) buffered_source.close
+        }
+
+        if (alluxio_master == null) {
+          throw new RuntimeException(
+            s"Can't find alluxio.master.hostname from $alluxio_home/conf/alluxio-site.properties.")
+        }
+        alluxioMasterHost = Some(alluxio_master + ":" + alluxio_port)
+        // load mounted point by call Alluxio mount command.
+        // We also can get from REST API http://alluxio_master:alluxio_web_port/api/v1/master/info.
+        val (ret, output) = runAlluxioCmd(" fs mount")
```

Comment:
I saw some exceptions will be thrown when failed to detect alluxio configuration or something else. But here if the "fs mount" failed to run, do we need to throw the exception also?

Reply:
Here I want to get the mounted path from alluxio command, but if failed to get, I regard it as no mounted path, and the exception will be threw when mounting a new path later.

Reply:
The alluxio command seq is like ("su", "ubuntu", "-c", "/opt/alluxio-2.8.0/bin/alluxio"), 
The string of " fs mount" is supposed to append to the last item in the seq to generate like ("su", "ubuntu", "-c", "/opt/alluxio-2.8.0/bin/alluxio fs mount").  
The original command should be  su ubuntu -c "/opt/alluxio-2.8.0/bin/alluxio fs mount".

Reply:
Couple of tests done in local workstation,here are my thoughts:
1. The error is like `java.lang.RuntimeException: Mount bucket s3a://mybucket/ to /mybucket failed 1`. Here the `1` is the `stdout` which does not give much debug info. I hope we can print stderr and stdout to show the reason why mount failed.
2. In my env, there is no need to "su" because the Alluxio cluster and Spark users are the same.
So if I tried to set:
```
spark.conf.set("spark.rapids.alluxio.cmd", "/home/xxx/alluxio-2.8.0/bin/alluxio")
```
Then it will fail with :
```
java.io.IOException: Cannot run program "/home/xxx/alluxio-2.8.0/bin/alluxio fs mount --readonly --option s3a.accessKeyId=xxx --option s3a.secretKey=yyy /mybucket s3a://mybucket/": error=2, No such file or directory
```

Reply:
I think the latest code has fixed above issue.

Reply:
I still think runAlluxioCmd should deal with putting a space in between it rather then having caller do it. we can do it later as followup though too

Reply:
Changed to add the space in runAlluxioCmd.

Reply:
@GaryShen2008 I just confirmed that now the issue regarding `spark.rapids.alluxio.cmd` got fixed based on my test on my local workstation. Now it does not need to `su`